R/matching.R

```diff
@@ -101,7 +115,13 @@ closest <- function(x, table, tolerance = Inf, ppm = 0,
     if (!is.numeric(nomatch) || length(nomatch) != 1L)
         stop("'nomatch' has to be a 'numeric' of length one.")
 
-    table <- table[!is.na(table)]
+    if (.check && (
+            !identical(FALSE, is.unsorted(x) ||
```

Comment:
why `identical` here and not simply `.check && (is.unsorted(x) || is.unsorted(table))` ?

Reply:
If `x` or `table` contain `NA`, `is.unsorted` return `NA` as well, e.g. `is.unsorted(c(1, NA))`. So I have to catch both cases which could easily be done with `identical`.

Reply:
OK, I see - but then, are you not missing a closing bracket? i.e. `!identical(FALSE, is.unsorted(x)) ||` instead of `!identical(FALSE, is.unsorted(x) ||`?

Reply:
Thanks, indeed.